staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity.go

```diff
@@ -37,31 +38,44 @@ type LazyErrorNodeSelector struct {
 	terms []nodeSelectorTerm
 }
 
-// NewNodeSelector returns a NodeSelector or all parsing errors found.
-func NewNodeSelector(ns *v1.NodeSelector) (*NodeSelector, error) {
-	lazy := NewLazyErrorNodeSelector(ns)
-	var errs []error
+// WithPath sets a field.Path to be used as root for parse errors.
+func WithPath(p *field.Path) ParseOption {
+	return func(o *parseOptions) {
+		o.path = p
+	}
+}
+
+// NewNodeSelector returns a NodeSelector or aggregate parsing errors found.
+func NewNodeSelector(ns *v1.NodeSelector, opts ...ParseOption) (*NodeSelector, error) {
+	lazy := NewLazyErrorNodeSelector(ns, opts...)
+	var errs field.ErrorList
 	for _, term := range lazy.terms {
-		if term.parseErr != nil {
-			errs = append(errs, term.parseErr)
+		if len(term.parseErrs) > 0 {
+			errs = append(errs, term.parseErrs...)
 		}
 	}
 	if len(errs) != 0 {
-		return nil, errors.NewAggregate(errs)
+		return nil, errs.ToAggregate()
 	}
 	return &NodeSelector{lazy: *lazy}, nil
 }
 
 // NewLazyErrorNodeSelector creates a NodeSelector that only reports parse
 // errors when no terms match.
-func NewLazyErrorNodeSelector(ns *v1.NodeSelector) *LazyErrorNodeSelector {
+func NewLazyErrorNodeSelector(ns *v1.NodeSelector, opts ...ParseOption) *LazyErrorNodeSelector {
+	o := parseOptions{}
+	for _, opt := range opts {
+		opt(&o)
+	}
 	parsedTerms := make([]nodeSelectorTerm, 0, len(ns.NodeSelectorTerms))
-	for _, term := range ns.NodeSelectorTerms {
+	path := o.path.Child("nodeSelectorTerms")
```

Comment:
`o.path` could be nil right?

Reply:
Good eye, but `(nil)`.Child actually works :)